src/spatial/detail/ArborX_AccessTraits.hpp

```diff
@@ -75,75 +71,45 @@ struct AccessTraits<
   using memory_space = typename View::memory_space;
 };
 
-namespace Details
+namespace Concepts
 {
 
-// archetypal alias for a 'memory_space' type member in access traits
-template <typename Traits>
-using AccessTraitsMemorySpaceArchetypeAlias = typename Traits::memory_space;
+template <class T, class U>
+concept not_same_as = !std::same_as<T, U>;
```

Comment:
I added this because there does not seem to be a `non-void` concept in std, and `!std::same_as<void>` did not work.